mailbox/tika/src/main/java/org/apache/james/mailbox/tika/TikaTextExtractor.java

```diff
@@ -50,19 +56,39 @@
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.MoreObjects;
 import com.google.common.base.Preconditions;
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.Weigher;
 import com.google.common.collect.ImmutableList;
 
 public class TikaTextExtractor implements TextExtractor {
 
+    private static final long CACHE_LIMIT_100_MB = 1024L * 1024L * 1024L * 100L;
+    private static final Duration CACHE_EVICTION_PERIOD = Duration.ofDays(1);
+
     private final MetricFactory metricFactory;
     private final TikaHttpClient tikaHttpClient;
     private final ObjectMapper objectMapper;
+    private final Cache<String, ParsedContent> cache;
```

Comment:
I don't think it's a good idea to put the cache in the TikaTextExtractor, it's invasive and prevent reuse.
What about write a CacheTextExtractor that would implement TextExtractor and delegate to any other TextExtractor ?